projects/wallet-template/src/background/rpc/chainspec.ts

```diff
@@ -65,5 +68,18 @@ export const removeChainSpecHandler: RpcMethodHandlers<
   BackgroundRpcSpec,
   Context
 >["removeChainSpec"] = async ([genesisHash], { lightClientPageHelper }) => {
-  await lightClientPageHelper.deleteChain(genesisHash)
+  if (genesisHash && wellKnownChainIdByGenesisHash[genesisHash]) {
+    throw new Error("cannot remove well-known chain")
+  }
+
+  const chains = await lightClientPageHelper.getChains()
+  const parachains = chains
+    .filter((chain) => chain.relayChainGenesisHash === genesisHash)
+    .map((chain) => chain.genesisHash)
+
+  await Promise.all(
+    [...parachains, genesisHash].map((hash) =>
+      lightClientPageHelper.deleteChain(hash),
+    ),
+  )
```

Comment:
I think this is covered in https://github.com/paritytech/substrate-connect/blob/fa0692522d4d06dba089b0907c278cde66b49fd1/packages/light-client-extension-helpers/src/background/background-helper.ts#L140-L161

Reply:
Oops good catch! Less work for the wallet :smiley: